### ISSUE TITLE
Use Swagger for the API

### ISSUE THREAD

**Issue by aadrian** [gitbucket/gitbucket#1603]
Please use [Swagger](http://swagger.io/) for the API:
Instead of just the error prone: https://github.com/gitbucket/gitbucket/wiki/API-WebHook 
something like http://petstore.swagger.io/  might be of great help (for testing too).

Using Swagger with Scalatra:
https://www.infoq.com/articles/swagger-scalatra/
https://abstractionextraction.wordpress.com/2013/03/31/using-swagger-with-scalatra/

thank you.

**Comment by michael-conrad** [comment]
Cross-reference: this request is being implemented in fork `michael-conrad/gitbucket#1` — spec and plan created for adding OpenAPI/Swagger spec endpoint via Scalatra SwaggerSupport 3.1.2.

Implementation plan: https://github.com/michael-conrad/gitbucket/issues/3

Approach uses `scalatra-swagger-javax` matching the existing Scalatra 3.1.2 `-javax` variant, with Jackson JSON support (`JacksonSwaggerBase`). Will report back once implementation is complete and merged.

**Comment by michael-conrad** [comment]
I've finaly got an initial working start on this. @takezoe is this something y'all would like to look at, it is mostly ai generated, but is largely rewriting the endpoints to register themselves into a swagger add-on. (ai agent grunt work - now that I can the swagger module actually working)

I can do a WIP PR if that would help, I'm focusing on trying to have one commit per controller file to simplify checking what was changed and to ensure scope creep or unwanted vibing isn't happening.

**Comment by michael-conrad** [comment]
I'm producing this for my own use, we've recently starting using ai agents + mcp and there is some serious issues in that regards with gitbucket. I'm starting off with the openapi spec for a custom client-side ai agent toolset - before moving on to seeing about generating a full MCP as a plugin.

**Comment by michael-conrad** [comment]
Picking this up — I'm working on wiring in Scalatra's built-in Swagger module (`scalatra-swagger-javax` 3.1.2) to serve a live OpenAPI 2.0 spec at `/api-docs/swagger.json`. Using the existing Scalatra dependency rather than a separate spec generator, so no new runtime framework.

The main wrinkle is GitBucket's `CompositeScalatraFilter` — Scalatra's auto-discovery doesn't work there, so registration has to be explicit. Working through that now.

Fork progress: michael-conrad/gitbucket#1
New issue for tracking: gitbucket/gitbucket#4017

🤖 Co-authored with AI: OpenCode (ollama-cloud/glm-5.1)
